src/Web/Slack/Experimental/Events/Types.hs

```diff
@@ -23,7 +23,7 @@ $(deriveJSON snakeCaseOptions ''ChannelType)
 
 -- | <https://api.slack.com/events/message>
 data MessageEvent = MessageEvent
-  { blocks :: [SlackBlock]
+  { blocks :: Maybe (NonEmpty SlackBlock)
```

Comment:
This is the issue you're having?

```
<missing_prop> -> Nothing
"blocks": null -> Nothing
"blocks": [] -> fail _
"blocks": [x1,...,xn] -> Just (x1 :| [x2,...,xn]
```

You're right that `NullableNonEmptyText` was a huge mistake. It might take a change to Aeson to fix things like this, but it's a common enough occurrence in APIs.

Have you considered `Maybe [SlackBlock]`? What is the advantage that `NonEmpty` is buying you here, as a consumer of this API (e.g. in MWB)?